addon/ActivityStreams.js

```diff
@@ -227,6 +172,117 @@ ActivityStreams.prototype = {
     });
   },
 
+  _setUpPageWorker(store) {
+    this._pageWorker = new PageWorker({store});
+    this._pageWorker.connect();
+    this._refreshAppState();
+  },
+
+  _initializePerfMeter() {
+    this._perfMeter = new PerfMeter(this.appURLs);
+  },
+
+  _initializeAppURLHider() {
+    this._appURLHider = new AppURLHider(this.appURLs);
+  },
+
+  _initializeMemoizer() {
+    this._memoizer = new Memoizer();
+    this._memoized = this._get_memoized(this._memoizer);
+  },
+
+  _initializeTelemetrySender() {
+    this._telemetrySender = new TelemetrySender();
+  },
+
+  _initializeBaselineRecommender(experimentProvider) {
+    // This is instantiated with a recommender based on weights if pref is true. Used to score highlights.
+    this._baselineRecommender = null;
+    if (experimentProvider.data.weightedHighlights) {
+      this._loadRecommender();
+    }
+  },
+
+  _initializePreviewProvier(experimentProvider, metadataStore, tabTracker) {
+    this._previewProvider = new PreviewProvider(tabTracker, metadataStore, experimentProvider);
+  },
+
+  _initializePrefProvider() {
+    this._prefsProvider = new PrefsProvider({
+      simplePrefs,
+      broadcast: this.broadcast.bind(this),
+      send: this.send.bind(this)
+    });
+    this._prefsProvider.init();
+  },
+
+  _initializeTabTracker(memoized, experimentProviderID) {
+    this._tabTracker = new TabTracker(
+      this.appURLs,
+      this.options.clientID,
+      memoized,
+      experimentProviderID
+    );
+  },
+
+  _initializeExperimentProvider() {
+    this._experimentProvider = new ExperimentProvider(
```

Comment:
You could actually initialize this in the constructor since has no side effects, and just call `_experimentProvider.init` directly.